commit 27e2b808542593b88f796d4ea5b8ddde8ca6bc5f
Author: Gavin Shan <gwshan@linux.vnet.ibm.com>
Date:   Thu Feb 16 10:22:34 2017 +1100

    pci/hotplug/pnv-php: Disable MSI and PCI device properly
    
    commit 49f4b08e61547a5ccd2db551d994c4503efe5666 upstream.
    
    pnv_php_disable_irq() can be called in two paths: Bailing path in
    pnv_php_enable_irq() or releasing slot. The MSI (or MSIx) interrupts
    is disabled unconditionally in pnv_php_disable_irq(). It's wrong
    because that might be enabled by drivers other than pnv-php.
    
    This disables MSI (or MSIx) interrupts and the PCI device only if
    it was enabled by pnv-php. In the error path of pnv_php_enable_irq(),
    we rely on the newly added parameter @disable_device. In the path
    of releasing slot, @pnv_php->irq is checked.
    
    Fixes: 360aebd85a4c ("drivers/pci/hotplug: Support surprise hotplug in powernv driver")
    Signed-off-by: Gavin Shan <gwshan@linux.vnet.ibm.com>
    Reviewed-by: Andrew Donnellan <andrew.donnellan@au1.ibm.com>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1291_27e2b808542593b88f796d4ea5b8ddde8ca6bc5f.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1291_27e2b808542593b88f796d4ea5b8ddde8ca6bc5f.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
